viewer-prototype/src/browser/trace-server-status.ts

```diff
@@ -105,40 +104,62 @@ export class TraceServerConnectionStatusContribution extends DefaultFrontendAppl
         }
     }
 
-    private statusbarId = 'trace-connection-status';
+    protected async startServer(): Promise<void> {
+        this.updateStatusBar({ text: '$(sync~spin) Trace server starting' });
+
+        try {
+            await this.withTimeout(this.traceServerConfigService.startTraceServer(this.path, this.port));
+        } catch {
+            this.messageService.error('Failed to start trace server.');
+            this.handleOffline();
```

Comment:
I think for start of the server, it will always timeout when the server is started successfully because the spawned process doesn't exit in that case. Calling handleOffline() will change indicate that the sever needs to be started, however the server is actually running.